src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-logic.h

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/clusters/push-av-stream-transport-server/constants.h>
+#include <app/clusters/push-av-stream-transport-server/push-av-stream-transport-delegate.h>
+#include <app/clusters/push-av-stream-transport-server/push-av-stream-transport-storage.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <protocols/interaction_model/StatusCode.h>
+#include <vector>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class PushAvStreamTransportServerLogic
+{
+public:
+    PushAvStreamTransportServerLogic(EndpointId aEndpoint, BitFlags<PushAvStreamTransport::Feature> aFeatures);
+    ~PushAvStreamTransportServerLogic();
+
+    void SetDelegate(EndpointId aEndpoint, PushAvStreamTransportDelegate * delegate)
+    {
+        mDelegate = delegate;
+        if (mDelegate == nullptr)
+        {
+            ChipLogError(Zcl, "Push AV Stream Transport: Delegate is null");
+            return;
+        }
+        mDelegate->SetEndpointId(aEndpoint);
+    }
+
+    enum class UpsertResultEnum : uint8_t
+    {
+        kInserted = 0x00,
+        kUpdated  = 0x01,
+    };
+
+    struct PushAVStreamTransportDeallocateCallbackContext
+    {
+        PushAvStreamTransportServerLogic * instance;
+        uint16_t connectionID;
+    };
+
+    EndpointId mEndpointId = kInvalidEndpointId;
+
+    std::vector<std::shared_ptr<PushAVStreamTransportDeallocateCallbackContext>> mTimerContexts;
```

Comment:
I do wonder whether we need shared_ptr, or whether these could be unique_ptr.  But either way.